patch/ps_command.go

```
@@ -0,0 +1,176 @@
+package patch
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strings"
+	"unicode/utf8"
+
+	"golang.org/x/text/encoding/simplifiedchinese"
+	"golang.org/x/text/transform"
+)
+
+func decodePowerShellOutput(input []byte) (string, error) {
+	if utf8Output := string(input); strings.HasPrefix(utf8Output, "\uFEFF") {
+		return strings.TrimPrefix(utf8Output, "\uFEFF"), nil
+	} else if utf8.Valid(input) {
+		return utf8Output, nil
+	}
+
+	// if not UTF-8，try GBK
+	gbkDecoder := simplifiedchinese.GBK.NewDecoder()
+	decodedOutput, _, err := transform.Bytes(gbkDecoder, input)
+	if err != nil {
+		return "", err
+	}
+	return string(decodedOutput), nil
+}
+
+func executePowerShellCommand(command string) (string, error) {
+	// Creating PowerShell commands to force UTF-8 output
+	cmd := exec.Command("powershell", "-Command",
+		"$OutputEncoding = [Console]::OutputEncoding = [Text.Encoding]::UTF8; "+command)
+
+	var stdout, stderr bytes.Buffer
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+
+	err := cmd.Run()
+	if err != nil {
+		decodedError, _ := decodePowerShellOutput(stderr.Bytes())
```

handle all errors, never use "_" to ignore an error